src/handlers/mod.rs

```
@@ -32,3 +36,25 @@ pub struct RpcQueryParams {
 pub struct SuccessResponse {
     status: String,
 }
+
+/// Checking rate limit for the request in the handler
+pub async fn handle_rate_limit(
+    state: State<Arc<AppState>>,
+    headers: HeaderMap,
+    connect_info: ConnectInfo<SocketAddr>,
+    path: MatchedPath,
+    project_id: Option<&str>,
+) -> Result<(), RateLimitExceeded> {
+    state
+        .rate_limit
+        .as_ref()
+        .unwrap()
+        .is_rate_limited(
+            path.as_str(),
+            &network::get_forwarded_ip(headers.clone())
+                .unwrap_or_else(|| connect_info.0.ip())
```

I think this defaulting to connecting IP is unsafe. If something breaks with the forwarding headers then we should _not_ be using the connecting IP as that's supposed to be the load balancer. We should at least be logging this as an error, not silently defaulting. And possibly we should not use a default at all and disable the rate limit or some other type of disable.

In Notify Server for the analytics if the forwarded header wasn't available I didn't lookup the connecting IP and recorded empty geo location values and emitted an `error!()`

---

What's the issue with local testing? Can't you set an `X-Forwarded` header?